Source/Core/Common/WorkQueueThread.h

```diff
@@ -20,89 +22,148 @@ class WorkQueueThread
 {
 public:
   WorkQueueThread() = default;
-  WorkQueueThread(std::function<void(T)> function) { Reset(std::move(function)); }
+  WorkQueueThread(const std::string_view name, std::function<void(T)> function)
+  {
+    Reset(name, std::move(function));
+  }
   ~WorkQueueThread() { Shutdown(); }
-  void Reset(std::function<void(T)> function)
+
+  // Shuts the current work thread down (if any) and starts a new thread with the given function
+  // Note: Some consumers of this API push items to the queue before starting the thread.
+  void Reset(const std::string_view name, std::function<void(T)> function)
   {
     Shutdown();
-    m_shutdown.Clear();
-    m_cancelled.Clear();
+    std::lock_guard lg(m_lock);
+    m_thread_name = name;
+    m_shutdown = false;
     m_function = std::move(function);
     m_thread = std::thread(&WorkQueueThread::ThreadLoop, this);
```

Comment:
Not convinced it would save much here but do want to call out the addition of the `jthread` in C++20.  It automatically has the concept of "shutting down" with a stop token and `request_stop()`.  It seems like it would be very similar code wise but in case I'm missing something, I thought I'd mention it.

Reply:
Not supported in clang...not sure we use that compiler but probably better to ignore regardless.

Reply:
We don't want to add any other types of concurrency or signalling other than the condition var.